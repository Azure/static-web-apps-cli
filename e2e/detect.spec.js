const fs = require("fs");
const path = require("path");
const detect = require("./detect");

function findFolders(basePath) {
  const entries = fs.readdirSync(basePath, { withFileTypes: true });
  return entries.filter((entry) => entry.isDirectory()).map((entry) => entry.name);
}

function findSamples() {
  const folders = ["samples/api", "samples/app", "samples/ssr"];
  let allSamples = [];

  for (const folder of folders) {
    let samples = findFolders(path.join(__dirname, folder));
    samples = samples.map((sample) => path.join(__dirname, folder, sample));
    allSamples = [...allSamples, ...samples];
  }

  return allSamples;
}

function createTest(folderPath, name) {
  it(`should detect from root framework ${name}`, async () => {
    process.chdir(folderPath);

    let result = await detect(".");
    // Fix windows paths
    result = result.replace(/\\/g, "/");
    expect(result).toMatchSnapshot();
  });
}

describe("framework detection", () => {
  it("should detect frameworks", async () => {
    process.chdir(__dirname);

    let result = await detect("./samples", 2);
    // Fix windows paths
    result = result.replace(/\\/g, "/");
    expect(result).toMatchInlineSnapshot(`
      "Detected api folders (6):
      - samples/api/dotnet (.NET)
      - samples/api/dotnet-csx (.NET)
      - samples/api/dotnet-isolated (.NET)
      - samples/api/node (Node.js)
      - samples/api/node-ts (Node.js, TypeScript)
      - samples/api/python (Python)
      Detected app folders (56):
      - samples/app/angular (Angular)
      - samples/app/angular-scully (Angular, Scully)
      - samples/app/astro (Astro)
      - samples/app/astro-alpine (Astro)
      - samples/app/astro-lit (Astro)
      - samples/app/astro-multiple (Astro)
      - samples/app/astro-preact (Astro)
      - samples/app/astro-react (Astro)
      - samples/app/astro-solid (Astro)
      - samples/app/astro-svelte (Astro)
      - samples/app/astro-vue (Astro)
      - samples/app/aurelia (Aurelia)
      - samples/app/blazor-wasm (Blazor WASM)
      - samples/app/brunch (Brunch)
      - samples/app/capacitor/www (Static HTML)
      - samples/app/docsify (Static HTML)
      - samples/app/docusaurus (React, Docusaurus)
      - samples/app/eleventy (Eleventy)
      - samples/app/elm (Elm)
      - samples/app/ember (Ember.js)
      - samples/app/flutter (Flutter)
      - samples/app/gridsome (Gridsome)
      - samples/app/hexo (Hexo)
      - samples/app/hugo (Hugo)
      - samples/app/ionic-angular (Angular)
      - samples/app/ionic-react (React)
      - samples/app/ionic-vue (Vue.js)
      - samples/app/jekyll (Jekyll)
      - samples/app/lektor (Lektor)
      - samples/app/lit (Lit)
      - samples/app/mdbook (mdBook)
      - samples/app/metalsmith (Metalsmith)
      - samples/app/middleman (Middleman)
      - samples/app/mkdocs (MkDocs)
      - samples/app/nuxtjs (Vue.js, Nuxt.js)
      - samples/app/pelican (Pelican)
      - samples/app/polymer (Polymer)
      - samples/app/preact (Preact)
      - samples/app/react (React)
      - samples/app/react-static (React, React-static)
      - samples/app/riot (Riot.js)
      - samples/app/sapper (Sapper)
      - samples/app/slate (Slate)
      - samples/app/solid (Solid)
      - samples/app/static (Static HTML)
      - samples/app/stencil (Stencil.js)
      - samples/app/svelte (Svelte)
      - samples/app/svelte-kit (SvelteKit)
      - samples/app/vitepress (VitePress)
      - samples/app/vue (Vue.js, Vite)
      - samples/app/vuepress/docs (VuePress)
      - samples/app/wintersmith (Wintersmith)
      - samples/app/zola (Zola)
      - samples/ssr/angular-universal (Angular)
      - samples/ssr/nextjs (React, Next.js)
      - samples/ssr/remix (Remix)
      Undetected folders (3):
      - samples/ssr/blazor-server
      - samples/ssr/marko
      - samples/ssr/meteor"
    `);
  });

  findSamples().forEach((sample) => createTest(sample, path.basename(sample)));
});
