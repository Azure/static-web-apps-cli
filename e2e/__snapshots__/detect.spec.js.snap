// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`framework detection should detect from root framework angular 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Angular)"
`;

exports[`framework detection should detect from root framework angular-scully 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Angular, Scully)"
`;

exports[`framework detection should detect from root framework angular-universal 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Angular)"
`;

exports[`framework detection should detect from root framework astro 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-alpine 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-lit 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-multiple 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-preact 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-react 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-solid 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-svelte 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework astro-vue 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Astro)"
`;

exports[`framework detection should detect from root framework aurelia 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Aurelia)"
`;

exports[`framework detection should detect from root framework blazor-server 1`] = `
"Detected api folders (0):
- 
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework blazor-wasm 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Blazor WASM)"
`;

exports[`framework detection should detect from root framework brunch 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Brunch)"
`;

exports[`framework detection should detect from root framework capacitor 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- www (Static HTML)"
`;

exports[`framework detection should detect from root framework docsify 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Static HTML)"
`;

exports[`framework detection should detect from root framework docusaurus 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (React, Docusaurus)"
`;

exports[`framework detection should detect from root framework dotnet 1`] = `
"Detected api folders (1):
- . (.NET)
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework dotnet-csx 1`] = `
"Detected api folders (1):
- . (.NET)
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework dotnet-isolated 1`] = `
"Detected api folders (1):
- . (.NET)
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework eleventy 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Eleventy)"
`;

exports[`framework detection should detect from root framework elm 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Elm)"
`;

exports[`framework detection should detect from root framework ember 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Ember.js)"
`;

exports[`framework detection should detect from root framework flutter 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Flutter)"
`;

exports[`framework detection should detect from root framework gridsome 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Gridsome)"
`;

exports[`framework detection should detect from root framework hexo 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Hexo)"
`;

exports[`framework detection should detect from root framework hugo 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Hugo)"
`;

exports[`framework detection should detect from root framework ionic-angular 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Angular)"
`;

exports[`framework detection should detect from root framework ionic-react 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (React)"
`;

exports[`framework detection should detect from root framework ionic-vue 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vue.js)"
`;

exports[`framework detection should detect from root framework jekyll 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Jekyll)"
`;

exports[`framework detection should detect from root framework lektor 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Lektor)"
`;

exports[`framework detection should detect from root framework lit 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Lit)"
`;

exports[`framework detection should detect from root framework marko 1`] = `
"Detected api folders (0):
- 
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework mdbook 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (mdBook)"
`;

exports[`framework detection should detect from root framework metalsmith 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Metalsmith)"
`;

exports[`framework detection should detect from root framework meteor 1`] = `
"Detected api folders (0):
- 
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework middleman 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Middleman)"
`;

exports[`framework detection should detect from root framework mkdocs 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (MkDocs)"
`;

exports[`framework detection should detect from root framework nextjs 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (React, Next.js)"
`;

exports[`framework detection should detect from root framework node 1`] = `
"Detected api folders (1):
- . (Node.js)
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework node-ts 1`] = `
"Detected api folders (1):
- . (Node.js, TypeScript)
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework nuxtjs 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vue.js, Nuxt.js)"
`;

exports[`framework detection should detect from root framework pelican 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Pelican)"
`;

exports[`framework detection should detect from root framework polymer 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Polymer)"
`;

exports[`framework detection should detect from root framework preact 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Preact)"
`;

exports[`framework detection should detect from root framework python 1`] = `
"Detected api folders (1):
- . (Python)
Detected app folders (0):
- "
`;

exports[`framework detection should detect from root framework react 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (React)"
`;

exports[`framework detection should detect from root framework react-static 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (React, React-static)"
`;

exports[`framework detection should detect from root framework remix 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Remix)"
`;

exports[`framework detection should detect from root framework riot 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Riot.js)"
`;

exports[`framework detection should detect from root framework sapper 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Sapper)"
`;

exports[`framework detection should detect from root framework slate 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Slate)"
`;

exports[`framework detection should detect from root framework solid 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework static 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Static HTML)"
`;

exports[`framework detection should detect from root framework stencil 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Stencil.js)"
`;

exports[`framework detection should detect from root framework svelte 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Svelte)"
`;

exports[`framework detection should detect from root framework svelte-kit 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (SvelteKit)"
`;

exports[`framework detection should detect from root framework vite-lit 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vite-preact 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vite-react 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vite-svelte 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vite-vanilla 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vite-vue 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vitepress 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (VitePress)"
`;

exports[`framework detection should detect from root framework vue 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Vite)"
`;

exports[`framework detection should detect from root framework vuepress 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- docs (VuePress)"
`;

exports[`framework detection should detect from root framework wintersmith 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Wintersmith)"
`;

exports[`framework detection should detect from root framework zola 1`] = `
"Detected api folders (0):
- 
Detected app folders (1):
- . (Zola)"
`;
